src/Avalonia.Controls/Platform/ITopLevelImplWithTextInputMethod.cs

```diff
@@ -6,6 +6,6 @@ namespace Avalonia.Controls.Platform
 {
     public interface ITopLevelImplWithTextInputMethod : ITopLevelImpl
     {
-        public ITextInputMethodImpl TextInputMethod { get; }
+        public ITextInputMethodImpl? TextInputMethod { get; }
```

Comment:
Looks a bit wrong. Why is it nullable, when ITopLevelImplWithTextInputMethod is special interface specifically to provide this property?

Reply:
Because `ValidatingTopLevel` uses an `as` cast here:

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ValidatingToplevel.cs#L131

@kekekeks is this correct, or should that be a throwing cast?

Reply:
It's nullable because it might be not available for this particular toplevel. E. g. if we are running on Linux without any configured IME